nifi-nar-bundles/nifi-cdc/nifi-cdc-mssql-bundle/nifi-cdc-mssql-processors/src/main/java/org/apache/nifi/cdc/mssql/event/TableCapturePlan.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.cdc.mssql.event;
+
+import org.apache.nifi.cdc.mssql.MSSQLCDCUtils;
+import org.apache.nifi.util.StringUtils;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+
+public class TableCapturePlan {
+
+    public enum PlanTypes{
+        CDC,
+        SNAPSHOT
+    }
+
+    public MSSQLTableInfo getTable() {
+        return table;
+    }
+
+    public int getRowLimit() {
+        return rowLimit;
+    }
+
+    public boolean getCaptureBaseline() {
+        return captureBaseline;
+    }
+
+    public Timestamp getMaxTime() {
+        return maxTime;
+    }
+
+    public PlanTypes getPlanType() {
+        return planType;
+    }
+
+    private MSSQLTableInfo table;
+    private int rowLimit;
+    private boolean captureBaseline;
+    private Timestamp maxTime;
+
+    private PlanTypes planType;
+
+    public TableCapturePlan(MSSQLTableInfo table, int rowLimit, boolean captureBaseline, String sTime){
+        this.table = table;
+
+        this.rowLimit = rowLimit;
+        this.captureBaseline = captureBaseline;
+
+        if(!StringUtils.isEmpty(sTime)){
+            this.maxTime = Timestamp.valueOf(sTime);
+        }
+    }
+
+    public void ComputeCapturePlan(Connection con, MSSQLCDCUtils mssqlcdcUtils) throws SQLException {
```

Comment:
Nitpick on leading capital letter in method name

Reply:
Agreed.